feathr_project/feathr/_feature_registry.py

```diff
@@ -838,3 +839,142 @@ def _list_registered_entities_with_details(self, project_name: str = None, entit
+        source_entity = self.purview_client.get_entity(guid=guid)["entities"][0]
+        return HdfsSource(name=source_entity["attributes"]["name"],
+                event_timestamp_column=source_entity["attributes"]["event_timestamp_column"],
+                timestamp_format=source_entity["attributes"]["timestamp_format"],
+                path=source_entity["attributes"]["path"],
+                registry_tags=source_entity["attributes"]["tags"]
+                )
+
+    def _get_features_by_guid(self, guid) -> List[FeatureAnchor]:
+        feature_entities = self.purview_client.get_entity(guid=guid)["entities"]
+        feature_list=[]
+        key_list = []
+        for feature_entity in feature_entities:
+            for key in feature_entity["attributes"]["key"]:
+                key_list.append(TypedKey(key_column=key["key_column"], key_column_type=key["key_column_type"], full_name=key["full_name"], description=key["description"], key_column_alias=key["key_column_alias"]))
+
+            # after get keys, put them in features
+            feature_list.append(Feature(name=feature_entity["attributes"]["name"],
+                    feature_type=None, # stored as a hocon string, can be parsed using pyhocon
+                    transform=None, #transform attributes are stored in a dict fashion , can be put in a WindowAggTransformation
+                    key=key_list,
+                    registry_tags=feature_entity["attributes"]["tags"],
+
+            ))
+        return feature_list 
+    def get_feature_by_fqdn_type(self, qualifiedName, typeName):
+        """
+        Get a single feature by it's QualifiedName and Type
+        Returns the feature else throws an AtlasException with 400 error code
+        """
+        response = self.purview_client.get_entity(qualifiedName=qualifiedName, typeName=typeName)
+        entities = response.get('entities')
+        for entity in entities:
+            if entity.get('typeName') == typeName and entity.get('attributes').get('qualifiedName') == qualifiedName: 
+                return entity
+       
+    def get_feature_by_fqdn(self, qualifiedName):
+        """
+        Get feature by qualifiedName
+        Returns the feature else throws an AtlasException with 400 error code
+        """        
+        guid = self.get_feature_guid(qualifiedName)
+        return self.get_feature_by_guid(guid)
+    
+    def get_feature_by_guid(self, guid):
+        """
+        Get a single feature by it's GUID
+        Returns the feature else throws an AtlasException with 400 error code
+        """ 
+        response = self.purview_client.get_single_entity(guid=guid)
+        return response
+    
+    def get_feature_lineage(self, guid):
+        """
+        Get feature's lineage by it's GUID
+        Returns the feature else throws an AtlasException with 400 error code
+        """
+        return self.purview_client.get_entity_lineage(guid=guid)
+
+    def get_feature_guid(self, qualifiedName):
```

Comment:
let's standardize. If it's ID, then we just say ID. GUID is a technical implementation details that we should not expose to users. We should use ID across the board and explain to user in the manual what's ID.

Reply:
changed